.github/workflows/rust-publish.yaml

```diff
@@ -0,0 +1,41 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Rust crate publish
+
+on:
+  create:
```

Comment:
**suggestion**: This should be similar to the other 

```suggestion
on:
  push:
    branches:
      - master
      - release-*
  pull_request:
```

Reply:
Not in this case, because we want to publish only  when someone creates a Git reference (Git branch or tag)in the repository.

Reply:
Sorry, this was ment for another file the `.github/workflows/rust-check-build.yaml`